src/python/WMCore/MicroService/MSUnmerged/MSUnmerged.py

```diff
@@ -425,35 +425,46 @@ def cleanRSE(self, rse):
 
         return rse
 
-    def _purgeTree(self, ctx, baseDirPfn):
+    def _purgeTree(self, ctx, baseDirPfn, isDirEntry=False):
         """
         A method to be used for purging the tree bellow a specific branch.
         It deletes every empty directory bellow that branch + the origin at the end.
-        :param ctx:  The gfal2 context object
-        :return:     Bool: True if it managed to purge everything, False otherwise
+        :param ctx:        The gfal2 context object
+        :param baseDirPfn: The base entry for starting the recursion
+        :param isDirEntry: Bool flag to avoid extra `stat` operations
+                           NOTE: When called from inside a recursion, we have already checked if the entry point is a directory
+        :return:           Bool: True if it managed to purge everything, False otherwise
         """
         # NOTE: It deletes only directories and does not try to unlink any file.
 
         # First test if baseDirPfn is actually a directory entry:
+        if not isDirEntry:
+            try:
+                entryStat = ctx.stat(baseDirPfn)
+                if not stat.S_ISDIR(entryStat.st_mode):
+                    self.logger.error("The base pfn: %s is not a directory entry.", baseDirPfn)
+                    return False
+            except gfal2.GError as gfalExc:
+                if gfalExc.code == errno.ENOENT:
+                    self.logger.warning("MISSING baseDir: %s", baseDirPfn)
+                    return True
```

Comment:
I do not know gfal codes and can't comment if return statement is correct here. Based on the code we issue warning about missing directory which kind of contradict logic if if block which should check if entry is a directory.

Reply:
The `.stat` method is also a valid method for checking for directory record presence. It throws `ENOENT` for missing entry, regardless if directory or file. So if we catch such an exception here we are sure the record on the file system is already missing.